ios/Fabric/Video/RCTVideo.h

```diff
@@ -0,0 +1,113 @@
+#import <AVFoundation/AVFoundation.h>
+#import "AVKit/AVKit.h"
+//#import "UIView+FindUIViewController.h"
+#import "RCTVideoPlayerViewController.h"
+#import "RCTVideoPlayerViewControllerDelegate.h"
+#import <React/RCTComponent.h>
+#import <React/RCTBridgeModule.h>
+
+#if __has_include(<react-native-video/RCTVideoCache.h>)
+#import <react-native-video/RCTVideoCache.h>
+#import <DVAssetLoaderDelegate/DVURLAsset.h>
+#import <DVAssetLoaderDelegate/DVAssetLoaderDelegate.h>
+#endif
+
+# pragma mark - video event delegate
+
+@protocol RCTVideoEventDelegate
+// TODO follow up all events in RCTVideo.swift
+//- (void)onVideoProgressWithCurrentTime:(NSNumber *)currentTime
+//       playableDuration:(NSNumber *)playableDuration
+//       seekableDuration:(NSNumber *)seekableDuration;
+//- (void)onVideoLoadWithCurrentTime:(NSNumber *)currentTime duration:(NSNumber *)duration naturalSize:(NSDictionary *)naturalSize;
+//- (void)onVideoLoadStartWithIsNetwork:(BOOL)isNetwork type:(NSString *)type uri:(NSString *)uri;
+//- (void)onVideoBufferWithIsBuffering:(BOOL)isBuffering;
+//- (void)onVideoErrorWithError:(NSDictionary *)error;
+//- (void)onGetLicenseWithLicenseUrl:(NSString *)licenseUrl contentId:(NSString*)contentId spcBase64:(NSString *)spcBase64;
+//- (void)onVideoSeekWithCurrentTime:(NSNumber *)currentTime seekTime:(NSNumber *)seekTime finished:(BOOL)finished;
+//- (void)onVideoEnd;
+//- (void)onTimedMetadata;
+//- (void)onVideoAudioBecomingNoisy;
+//- (void)onVideoFullscreenPlayerWillPresent;
+//- (void)onVideoFullscreenPlayerDidPresent;
+//- (void)onVideoFullscreenPlayerWillDismiss;
+//- (void)onVideoFullscreenPlayerDidDismiss;
+//- (void)onReadyForDisplay;
+//- (void)onRestoreUserInterfaceForPictureInPictureStop;
+//- (void)onPictureInPictureStatusChangedWithIsActive:(BOOL)isActive;
+//- (void)onPlaybackRateChangeWithPlaybackRate:(NSNumber *)playbackRate;
+//- (void)onVideoExternalPlaybackChangeWithIsExternalPlaybackActive:(BOOL)isExternalPlaybackActive;
+//- (void)onReceiveAdEventWithEvent:(NSString *)event;
+
+@end
+
+#if __has_include(<react-native-video/RCTVideoCache.h>)
+@interface RCTVideo : UIView <RCTVideoPlayerViewControllerDelegate, DVAssetLoaderDelegatesDelegate, AVAssetResourceLoaderDelegate>
```

Comment:
Shouldn't this one also contain `AVPictureInPictureControllerDelegate` ?

Reply:
Yes I can find out `AVPictureInPictureControllerDelegate` in RCTPictureInPicture.swift
I will add it!

Reply:
@KrzysztofMoch Hmm, I am not sure.
That RCTVideo interface is for `#if __has_include(<react-native-video/RCTVideoCache.h>)`
Do we need to extends `AVPictureInPictureControllerDelegate` for RCTVideoCache too?

There is
```objc
#else
@interface RCTVideo : UIView <RCTVideoPlayerViewControllerDelegate, AVPictureInPictureControllerDelegate, AVAssetResourceLoaderDelegate>
#endif
```
which has extended `AVPictureInPictureControllerDelegate` in the last endif.

Reply:
I think we should - otherwise we won't be able to use PIP when cache is enabled (I think)

Reply:
Okay.
added extend AVPictureInPictureControllerDelegate when cache is enabled!
[feat: 🎨 extends AVPictureInPictureControllerDelegate when cache is en…](https://github.com/TheWidlarzGroup/react-native-video/pull/3965/commits/5fa3c095d1834a73125448a937cf74fd54857b28)